### PR TITLE
Feature/#701 - DataNode._get_user_properties() API

### PR DESCRIPTION
**Pull request by trgiangdo**
https://github.com/Avaiga/taipy-core/issues/701

### CODE REVIEW

**Review by jrobinAV — CHANGES_REQUESTED** [review]
Hello, 
That's not the requirement of ticket #701.

The purpose is to return all properties provided by the end user, without the ones created by Taipy. The list of properties to remove is available [here](https://docs.google.com/document/d/1L5_OPxiAZX9ZU9d6yomMzDZV-Bpv6bA-SW_GRC7n8Do/edit#heading=h.rjc1l3svrrf1).

### Properties to remove: 
**For csv:** exposed_type, path, default_path, default_data, has_header
**For Excel:** exposed_type, path, default_path, default_data, has_header, sheet_name
**For Generic:** read_fct, read_fct_args, write_fct, write_fct_args
**For In Memory:** default_data
**For JSON:** path, default_path, default_data, encoder, decoder
**For Mongo:** custom_document, db_name, collection_name, db_extra_args, db_host, db_port, db_username ,db_password
**For Parquet:** exposed_type, path, default_path, default_data, engine, compression, read_kwargs, write_kwargs
**For Pickle:** path, default_path, default_data, is_generated
**For SQL:** read_query, write_query_builder, exposed_type, db_engine, db_username, db_password, db_name, db_host, db_port, db_driver, db_extra_args, sqlite_folder_path, sqlite_file_extension
**For sql_table:** table_name, db_engine, exposed_type, db_username, db_password, db_name, db_host, db_port, db_driver, db_extra_args, sqlite_folder_path, sqlite_file_extension

**Comment by trgiangdo** [comment]
Sorry I misunderstood the ticket since there is a `*list_of_properties_to_keep` parameter.

So the `DataNode._get_custom_properties()` will not accept any parameter at all, right?

File: src/taipy/core/data/data_node.py [hunk 1]
```diff
@@ -267,6 +269,10 @@ def properties(self):
         self._properties = _Reloader()._reload(self._MANAGER_NAME, self)._properties
         return self._properties
 
+    def _get_custom_properties(self) -> Dict[str, Any]:
```
Comment (jrobinAV) [inline src/taipy/core/data/data_node.py]: I am changing my mind, but what about renaming custom properties to user properties `_get_user_properties`?

File: src/taipy/core/data/data_node.py [hunk 2]
```diff
@@ -267,6 +269,10 @@ def properties(self):
...
+        """Get custom properties of the data node except from the ones created by TAIPY."""
```
Comment (jrobinAV) [inline src/taipy/core/data/data_node.py]:
```suggestion
        """ Get user properties."""
```

**Review by jrobinAV — DISMISSED** [review]
minor suggestion on the name

File: src/taipy/core/data/data_node.py (hunk 1 above)
Comment (trgiangdo) [reply src/taipy/core/data/data_node.py]: It's an internal API for now, so why not 😂

Comment (trgiangdo) [reply src/taipy/core/data/data_node.py]:
```suggestion
    def _get_user_properties(self) -> Dict[str, Any]:
```

**Review by jrobinAV — APPROVED** [review]

**Pull request merged**